server/routes/nl.py

```diff
@@ -101,21 +108,41 @@ def _remove_places(query, places_found):
 
 def _infer_place_dcid(places_found):
   if not places_found:
+    logging.info("places_found is empty. Nothing to retrieve from Maps API.")
     return ""
 
   place_dcid = ""
-  place = _maps_place(places_found[0])
-  # If maps API returned a valid place, use the place_id to
-  # get the dcid.
-  if place and ("place_id" in place):
-    place_id = place["place_id"]
-    logging.info(f"MAPS API found place with place_id: {place_id}")
-    place_ids_map = _dc_recon([place_id])
-
-    if place_id in place_ids_map:
-      place_dcid = place_ids_map[place_id]
-
-  logging.info(f"DC API found DCID: {place_dcid}")
+  for p_str in places_found:
+    # If this is a special place, return the known DCID.
+    if p_str.lower() in constants.SPECIAL_PLACES_TO_GEOIDS:
+      place_dcid = constants.SPECIAL_PLACES_TO_GEOIDS[p_str.lower()]
+      logging.info(
+          f"{p_str} was found in SPECIAL_PLACES_TO_GEOIDS. Returning its DICD {place_dcid} without querying Maps API."
+      )
+      break
+
+    # Iterate over all the places until a valid place DCID is found.
```

Comment:
nit: This comment makes more sense in line 115 (where the for loop is there)?

Reply:
done